### ISSUE TITLE
[distributed] Unable to pass a host name which resolves to 127.0.0.1 when running a Minio instance

### ISSUE THREAD

**Issue by vadmeste** [minio/minio#4518]
## Expected Behavior
Running one instance should work, we target distributed Minio. 

## Current Behavior
Cannot run the instance, I have the following error: 
```
FATA[0000] Invalid command line arguments server=‘:9000’, args=[https://minio1/tmp/vadmeste/export1 https://minio1/tmp/vadmeste/export2 https://minio2/tmp/vadmeste/export3 https://minio2/tmp/vadmeste/export4]  cause=no endpoint found for this host source=[server-main.go:144:serverHandleCmdArgs()]
```

In the same machine, minio1 is recognized and it points to 127.0.0.1

```
$ ping minio1 -c1
PING minio1 (127.0.1.1) 56(84) bytes of data.
64 bytes from minio1 (127.0.1.1): icmp_seq=1 ttl=64 time=0.087 ms

--- minio1 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.087/0.087/0.087/0.000 ms
```

## Possible Solution
A workaround for the user is to make minio1 hostname points to a public and not to a local ip. However, I am not sure why we error out in this case, besides the error message is not exact. (`no endpoint found for this host`)

## Steps to Reproduce (for bugs)
```
$ ./minio server https://minio1/tmp/vadmeste/export1 https://minio1/tmp/vadmeste/export2 https://minio2/tmp/vadmeste/export3 https://minio2/tmp/vadmeste/export4
```

## Context
Testing

## Your Environment
master

**Comment by harshavardhana** [comment]
The case you are mentioning on the same node is addressed by you running Minio on separate ports and specifying them properly.

What you are essentially doing here is running standalone Erasure not distributed that's why it's not supported since you might be doing something wrong by specifing hosts rather than disks.

**Comment by vadmeste** [comment]
> What you are essentially doing here is running standalone Erasure not distributed

How standalone? I specified both minio1 and minio2 in arguments

**Comment by balamurugana** [comment]
@vadmeste 

> $ ./minio server https://minio1/tmp/vadmeste/export1 https://minio1/tmp/vadmeste/export2 https://minio2/tmp/vadmeste/export3 https://minio2/tmp/vadmeste/export4

What setup type are you trying to do ie XL or DistXL?
DistXL requires non-loopback IPs of node(s) and XL requires either URL style endpoints pointing to local host or Path style endpoints.

**Comment by vadmeste** [comment]
> What setup type are you trying to do ie XL or DistXL?

I want DistXL. Two nodes minio1 and minio2, each node has two disks.
In my case, minio1 points to 127.0.0.1 and minio2 points to 65.x.x.x

> DistXL requires non-loopback IPs of node(s)

Any reason for this requirement?

In all cases, the error message is not correct, it says `no endpoint found for this host` though it should say `a loop-back address should not be specified`

**Comment by harshavardhana** [comment]
> I want DistXL. Two nodes minio1 and minio2, each node has two disks.
> In my case, minio1 points to 127.0.0.1 and minio2 points to 65.x.x.x

This is not allowed @vadmeste while it is an acceptable case, it usually points to a wrongly configured DNS.  So we better fail.

**Comment by balamurugana** [comment]
> I want DistXL. Two nodes minio1 and minio2, each node has two disks.
> In my case, minio1 points to 127.0.0.1 and minio2 points to 65.x.x.x
> 

Its cautious decision is not to support `127.x.x.x` where these endpoints have to be compared to validate the setup at rpc login.  Currently its not done yet.

>>     DistXL requires non-loopback IPs of node(s)
> 
> Any reason for this requirement?
>

They can be any IPs of local host.  Sorry for the confusion.

> In all cases, the error message is not correct, it says no endpoint found for this host though it should say a loop-back address should not be specified

Yep. It can be improved.

**Comment by krishnasrinivas** [comment]
Fixed by https://github.com/minio/minio/pull/4105

**Comment by harshavardhana** [comment]
Closing as fixed by #4105
